Cleanup and modernize udev-extraconf and niacctbase

amstewart opened the pull request:
This patchset performs the following substantial changes:
* generally cleanup the `udev-extraconf` recipe
* move the `ni` udev rules from `udev-extraconf` to a new subpackage`niacctbase-udev`, so that the `udev-extraconf` recipe no longer contains NI-specific logic.
* generally cleanup the `niacctbase` recipe.
* Bump the package versions of both `udev-extraconf` and `niacctbase` to `1.1`.
* Switch both recipes to use the new variable override syntax (`:`).
* Improve the package metadata variables in both recipes.

Nothing in this patchset is intended to change the runtime behavior of NILRT. This is just a major recipe refactor.

# Testing
Created a NIRLT VM from this patchset and compared the layout and permissions of the `/dev` tree, the udev rules, and the installed packages against the latest 22.8 export. The VMs do not generate *every* device that can appear on supported hardware, but I was able to verify that there are no changes in what is present on VMs. `niacctbase-udev` is now installed by default.

amstewart committed:
udev-extraconf: whitespace and styling

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
udev-extraconf: remove nilrt-nxg logic

The `nilrt-nxg` distro flavor has been deprecated.

Remove recipe content referencing that distribution.

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
udev-extraconf: remove ARM content

ARM isn't a supported architecture in the hardknott mainline of NILRT.

Remove recipe content which references ARM.

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
udev-extraconf: add package metadata

Add package metadata for HOMEPAGE and BUGTRACKER.

Also reorder PACKAGE_ARCH toward the end of the recipe, for styling.

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
udev-extraconf: cleanup do_install

Cleanup the do_install function to be a little cleaner.

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
udev-extraconf: further refine do_install

Move rules.d and scripts into their own source directories for
cleanliness, and to slightly abbreviate the do_install logic.

Also, apply the ordinal-prefixes to some of the rules within the
do_install, rather than having them be a static part of the filename.
How the files are ordered is packaging/sysadmin logic, more-so than
being intrinsic properties of the file.

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
move ni group udev rules to niacctbase

The `localextra.rules` and `localextra_rfkill.rules` definitions contain
some ni-specific logic to give the `ni` user group access to certain
kernel devices.

Move this logic to the `niacctbase` recipe directly, so that
udev-extraconf can remain relatively distro-agnostic.

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
niacctbase: assert a package version

The niacctbase recipe doesn't explicitly define a package version, and
so gets the default value of `1.0`. To better track when we make
changes, we should assert a version within the recipe.

Add an explicit `PV` and set it to 1.1, to note that substantial changes
have been made to the recipe contents.

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
niacctbase: remove nilrt-nxg logic

Meta-nilrt no longer supports a `nilrt-nxg` distro flavor. Remove logic
which references this distro.

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
udev-extraconf: use new variable override syntax

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
udev-extraconf: remove unneeded dependencies

Now that all `ni` group logic has been moved to the `niacctbase` recipe,
the `udev-extraconf` recipe no longer needs a build- or run-time
dependency on `niacctbase` or the associated shadow and pseudo deps.

Remove them for cleanliness.

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
niacctbase: enhance package metadata

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
niacctbase: use new variable override syntax

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
udev-extraconf: bump package version

Bump the udev-extraconf package version, since we have made substantive
changes to the package contents since version 1.0.

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
niacctbase: create mutual RDEPENDS on -udev

Create a mutual RDEPENDS relationship between the niacctbase and the
niacctbase-udev subpackages, to ensure that they are always both
installed.

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
udev-extraconf: update hotplug script location

The hotplug udev script has changed file extension slightly.

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

shruthi-ravi commented on recipes-ni/niacctbase/niacctbase.bb:
```diff
@@ -1,4 +1,7 @@
-DESCRIPTION = "Basic user and group settings for NI Linux RT applications."
+SUMMARY = "Common user and group settings for NI applications"
```
Should it be `applications and devices` based on your most recent changes?

amstewart replied:
Done.

amstewart committed:
niacctbase: adjust package SUMMARY

niacctbase owns permission configuration of NI kernel devices as well
(though maybe it shouldn't.) Adjust the package summary to express this
purpose.

Signed-off-by: Alex Stewart <alex.stewart@ni.com>
